appinventor/components/src/com/google/appinventor/components/runtime/SpeechRecognizerController.java

```diff
@@ -0,0 +1,7 @@
+package com.google.appinventor.components.runtime;
+
+abstract class SpeechRecognizerController {
```

Comment:
The logic of making this an abstract class would be to include the addListener logic here and not repeat it in the individual implementations. Either do that or make this an interface.